modules/overrides/overrides.go

```diff
@@ -123,6 +136,65 @@ func (o *Overrides) stopping(_ error) error {
 	return nil
 }
 
+func (o *Overrides) tenantOverrides() *perTenantOverrides {
+	if o.runtimeConfigMgr == nil {
+		return nil
+	}
+	cfg, ok := o.runtimeConfigMgr.GetConfig().(*perTenantOverrides)
+	if !ok || cfg == nil {
+		return nil
+	}
+
+	return cfg
+}
+
+func (o *Overrides) Handler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var tenantOverrides perTenantOverrides
+		if o.tenantOverrides() != nil {
+			tenantOverrides = *o.tenantOverrides()
+		}
+		var output interface{}
+		cfg := Config{
+			Defaults:           o.defaultLimits,
+			PerTenantOverrides: tenantOverrides,
+		}
+		switch r.URL.Query().Get("mode") {
+		case "diff":
+			// Default runtime config is just empty struct, but to make diff work,
+			// we set defaultLimits for every tenant that exists in runtime config.
+			defaultCfg := perTenantOverrides{TenantLimits: map[string]*Limits{}}
+			defaultCfg.TenantLimits = map[string]*Limits{}
+			for k, v := range tenantOverrides.TenantLimits {
```

Comment:
I think this will result in null pointer dereference if `o.tenantOverrides == nil`

Reply:
I think it's fine since we're instantiating a zero value in line 153 and only assign a value if `o.tenantOverrides() != nil`